README.adoc

```diff
@@ -226,4 +175,79 @@ You must set the `hibernate.hql.bulk_id_strategy` setting in `hibernate.properti
 This is because Hibernate's default behavior (`PersistentTableBulkIdStrategy`) attempts to create intermediate tables to handle delete and update operations on the multiple tables that represent a `JOINED` inheritance hierarchy, but these table creations statements do not conform to Cloud Spanner DDL.
 Using one of the `Inline` bulk-ID strategy classes given above resolves this issue.
 
+== Cloud Spanner Hibernate ORM Features
+
+The Cloud Spanner Hibernate Dialect supports most of the standard Hibernate and Java Persistence annotations, but there are important differences in features because of differences in https://cloud.google.com/spanner/docs/schema-and-data-model[Cloud Spanner's data model] from traditional SQL databases.
+
+The following sections below describe the feature set of the Cloud Spanner Hibernate Dialect as well as its limitations.
+
+[options="header"]
+|===
+| Unsupported Feature | Description
+| Constraints | No support for `FOREIGN KEY`, `UNIQUE`, or `ON DELETE CASCADE`. Cloud Spanner does not enforce relationship links outside of Hibernate.
+| Catalog and schema scoping for table names | Tables name references cannot contain periods or other punctuation.
+| Column default values | Cloud Spanner treats `null` like any other value and it gets no special handling.
+| Big-decimal or arbitrary-precision numbers | The dialect does not support `java.math.BigDecimal` because Cloud Spanner doesn't have SQL `NUMERIC` or `DECIMAL` support.
+| Schema Generation for Interleaved Tables | The dialect currently does not support creating interleaved tables during schema generation.
+|===
+
+=== Constraints
+
+Cloud Spanner does not support database constraints.
+As a result, `SpannerDialect` does not currently support any constraints such as `FOREIGN KEY`, `UNIQUE`, or `ON DELETE CASCADE`.
+
+NOTE: Hibernate doesn't directly rely on the existence of constraints to perform its operations and leaves the enforcement of relationship links to the database.
+
+The lack of the foreign key constraint affects relationships and collection properties annotated with `@ElementCollection`.
```

Comment:
How does it affect?

Reply:
I don't know; @ChengyuanZhao - Could you help fill us in on this one?

Reply:
This is just mentioning how Hibernate uses FK constraints in the DDL for both relationships as well as collection fields, so not having FKs means both of these would just be performed as joins on otherwise unrelated columns. There might be some performance impact from that.

Reply:
OK I see.
```suggestion
The lack of the foreign key constraint applies to relationships and collection properties annotated with `@ElementCollection`.
```